### PR TITLE
implement request/response allOf checkers

### PR DESCRIPTION
**Pull request by ciprian-tibulca**

### CODE REVIEW

File: checker/check-request-property-all-of-updated.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			modifiedMediaTypes := operationItem.RequestBodyDiff.ContentDiff.MediaTypeModified
+			for _, mediaTypeDiff := range modifiedMediaTypes {
+				if mediaTypeDiff.SchemaDiff == nil {
+					continue
+				}
+
+				if mediaTypeDiff.SchemaDiff.AllOfDiff != nil && mediaTypeDiff.SchemaDiff.AllOfDiff.Added > 0 {
+					result = append(result, ApiChange{
+						Id:    "request-body-all-of-added",
+						Level: ERR,
```
Comment (ciprian-tibulca) [inline checker/check-request-property-all-of-updated.go:37]: breaking change

File: checker/check-request-property-all-of-updated.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+						OperationId: operationItem.Revision.OperationID,
+						Path:        path,
+						Source:      source,
+					})
+				}
+
+				if mediaTypeDiff.SchemaDiff.AllOfDiff != nil && mediaTypeDiff.SchemaDiff.AllOfDiff.Deleted > 0 {
+					result = append(result, ApiChange{
+						Id:    "request-body-all-of-removed",
+						Level: ERR,
```
Comment (ciprian-tibulca) [inline checker/check-request-property-all-of-updated.go]: breaking change

File: checker/check-request-property-all-of-updated.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+					mediaTypeDiff.SchemaDiff,
+					func(propertyPath string, propertyName string, propertyDiff *diff.SchemaDiff, parent *diff.SchemaDiff) {
+						if propertyDiff.AllOfDiff == nil {
+							return
+						}
+
+						if propertyDiff.AllOfDiff.Added > 0 {
+							result = append(result, ApiChange{
+								Id:    "request-property-all-of-added",
+								Level: ERR,
```
Comment (ciprian-tibulca) [inline checker/check-request-property-all-of-updated.go:72]: breaking change

File: checker/check-request-property-all-of-updated.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+								OperationId: operationItem.Revision.OperationID,
+								Path:        path,
+								Source:      source,
+							})
+						}
+
+						if propertyDiff.AllOfDiff.Deleted > 0 {
+							result = append(result, ApiChange{
+								Id:    "request-property-all-of-removed",
+								Level: ERR,
```
Comment (ciprian-tibulca) [inline checker/check-request-property-all-of-updated.go]: breaking change

**Review by blva — APPROVED** [review]
LGTM

File: BREAKING-CHANGES-EXAMPLES.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Examples of Breaking and Non-Breaking Changes
 These examples are automatically generated from unit tests.
 ## Examples of breaking changes
+[adding 'allOf' schema from the request body or request body property is breaking](checker/checker_breaking_test.go?plain=1#L654)
```
Comment (reuvenharrison) [inline BREAKING-CHANGES-EXAMPLES.md]: Should be: "to the request body"

File: checker/check-request-property-all-of-updated.go (hunk 2 above)
Comment (reuvenharrison) [reply checker/check-request-property-all-of-updated.go]: I don't understand why this is breaking:
Suppose a client sends a request body and it matched "All Of" the schemas in the spec, now we remove one of the allOf specs, the same request body will still match.
Am I missing something?

File: BREAKING-CHANGES-EXAMPLES.md (hunk 5 above)
Comment (ciprian-tibulca) [reply BREAKING-CHANGES-EXAMPLES.md]: updated, thank you

File: checker/check-request-property-all-of-updated.go (hunk 2 above)
Comment (ciprian-tibulca) [reply checker/check-request-property-all-of-updated.go]: you are right, the request body will still match if one of the `allOf` specs is removed.
I changed to WARN and the BC message  to `// BC: removing 'allOf' schema from the request body or request body property is breaking with warn`, similar (level + BC message) to what we have for `request-property-removed`.
They are very similar, in the case of `allOf` we remove one or more properties from the request.
WDYT?

Comment (reuvenharrison) [reply checker/check-request-property-all-of-updated.go]: I see what you mean, and I agree with your solution, but I'd like to clarify the explanation for the record:
Removing a schema from allOf, **could** result in the effective removal of a property, and as such, it should be treated like request-property-removed.
However, removing a schema from allOf, doesn't necessarily result in the removal of a property, for example when the 2nd object is removed here:
```
schema:
  allOf:
    - type: object
      properties:
        prop1:
          type: string
          description: Some description
    - type: object
      properties:
        prop1:
          type: string
          maxLength: 30
```

**Review by reuvenharrison — APPROVED** [review]

**Pull request merged**